drivers/system_pathways/app/models/system_pathways/time_chart/race_and_ethnicity.rb

```diff
@@ -0,0 +1,193 @@
+        race_data = pluck_to_hash(race_columns.except('race_none', 'multi_racial').map { |k, v| [sp_c_t[k], v] }.to_h.merge(stay_length_col => 'Days to Move-In'), scope)
+        race_ethnicity_combinations.each do |key, value|
+          data = lookups[key]&.call(race_data, stay_length_col)
+          counts[value] = average(data.sum, data.count).round
+        end
+        [
+          p_label,
+          counts,
+        ]
+      end.to_h
+      e_counts[:ph_counts] = ph_counts
+
+      label = 'Served by Homeless System'
+      counts = {}
+      race_ethnicity_combinations.each_value do |k|
+        counts[k] ||= 0
+      end
+      # Get rid of the distinct from node_clients
+      stay_length_col = sp_c_t[:days_to_return]
+      scope = SystemPathways::Enrollment.joins(:client).where(id: node_clients(label).select(:id)).where(stay_length_col.not_eq(nil))
+      race_data = pluck_to_hash(race_columns.except('race_none', 'multi_racial').map { |k, v| [sp_c_t[k], v] }.to_h.merge(stay_length_col => 'Days to Return'), scope)
+      race_ethnicity_combinations.each do |key, value|
+        data = lookups[key]&.call(race_data, stay_length_col)
+        counts[value] = average(data.sum, data.count).round
+      end
+      [
+        label,
+        counts,
+      ]
+      e_counts[:return_counts] = counts
+    end
+  end
+
+  private def single_race(race_data, race, stay_length_col)
+    race_data.select { |client| client[sp_c_t[race]] && !client[sp_c_t['hispanic_latinaeo']] }.
+      select { |client| client.except(sp_c_t['id'], sp_c_t['hispanic_latinaeo'], stay_length_col).values.count(true) == 1 }.
+      map { |client| client[stay_length_col] }
+  end
+
+  private def single_race_latinaeo(race_data, race, stay_length_col)
+    race_data.select { |client| client[sp_c_t[race]] && client[sp_c_t['hispanic_latinaeo']] }.
+      select { |client| client.except(sp_c_t['id'], sp_c_t['hispanic_latinaeo']).values.count(true) == 1 }.
+      map { |client| client[stay_length_col] }
+  end
+
+  private def lookups
+    {
+      am_ind_ak_native: ->(race_data, stay_length_col) { single_race(race_data, 'am_ind_ak_native', stay_length_col) },
+      am_ind_ak_native_hispanic_latinaeo: ->(race_data, stay_length_col) { single_race_latinaeo(race_data, 'am_ind_ak_native', stay_length_col) },
+      asian: ->(race_data, stay_length_col) { single_race(race_data, 'asian', stay_length_col) },
+      asian_hispanic_latinaeo: ->(race_data, stay_length_col) { single_race_latinaeo(race_data, 'asian', stay_length_col) },
+      black_af_american: ->(race_data, stay_length_col) { single_race(race_data, 'black_af_american', stay_length_col) },
+      black_af_american_hispanic_latinaeo: ->(race_data, stay_length_col) { single_race_latinaeo(race_data, 'black_af_american', stay_length_col) },
+      hispanic_latinaeo: ->(race_data, stay_length_col) do
```

Comment:
Same here, let's group this with `multi_racial` and add a comment about what's going on in the select.